src/vm/jitinterface.cpp

```diff
@@ -5459,8 +5463,11 @@ void CEEInfo::getCallInfo(
         pResult->kind = CORINFO_VIRTUALCALL_LDVIRTFTN;  // stub dispatch can't handle generic method calls yet
         pResult->nullInstanceCheck = TRUE;
     }
-    // Non-interface dispatches go through the vtable
-    else if (!pTargetMD->IsInterface() && !IsReadyToRunCompilation())
+    // Non-interface dispatches go through the vtable.
+    // We'll special virtual calls to target methods in the corelib assembly when compiling in R2R mode and generate fragile-NI-like callsites for improved performance. We
+    // can do that because today we'll always service the corelib assembly and the runtime in one bundle. Any caller in the corelib version bubble can benefit from this
+    // performance optimization.
+    else if (!pTargetMD->IsInterface() && (!IsReadyToRunCompilation() || callerAndCalleeInSystemVersionBubble))
```

Comment:
Maybe create a small function `CallerAndCalleeInSystemVersionBubble` and call it here instead?

Right now, we are computing the `callerAndCalleeInSystemVersionBubble` value for number of cases that it is not needed.

Reply:
Sounds good